### ISSUE TITLE
Provide an overload of DeviceClient's ReceiveAsync taking in a cancellation token

### ISSUE THREAD

**Issue by Praburaj** [Azure/azure-iot-sdks#672]
It will be convenient to wait on a `CancellationToken` instead of a `TimeSpan`.

**Comment by tameraw** [comment]
Thanks for the feedback. We'll consider adding this in a future release. However, we happily accept contributions if you decide to submit a pull request for this.
https://github.com/Azure/azure-iot-sdks/blob/master/CONTRIBUTING.md

**Comment by tameraw** [comment]
Unfortunately, this is currently not a priority in our backlog. 
But if this is still a priority on your end, the Microsoft Customer Support team will try and help out on a best effort basis. To engage Microsoft support, you can create a support ticket directly from the [Azure Portal](https://ms.portal.azure.com/#blade/Microsoft_Azure_Support/HelpAndSupportBlade).
